packages/legend-application-studio/src/components/project-viewer/ProjectViewer.tsx

```diff
@@ -329,8 +330,10 @@ export const ProjectViewer = withEditorStore(
                       {...sideBarCollapsiblePanelGroupProps.remainingPanel}
                       minSize={300}
                     >
-                      {editorStore.isInFormMode && <EditPanel />}
-                      {editorStore.isInGrammarTextMode && <GrammarTextEditor />}
+                      {editorStore.graphEditorMode.mode ===
```

Comment:
I'm thinking should we shift our thinking here a bit, technically, everything is still `EditPanel`, but we have 1 tab (non-closable) instead. What do you think? We can do the same for `Editor.tsx`, this is a clean approach since it's purely `state-driven`